src/libexpr/flake/lockfile.hh

```diff
@@ -55,16 +55,14 @@ struct LockFile
     ref<Node> root = make_ref<Node>();
 
     LockFile() {};
-    LockFile(const nlohmann::json & json, const Path & path);
+    LockFile(std::string_view contents, std::string_view path);
```

Comment:
```suggestion
    LockFile(nlohmann::json json, std::string_view path);
```
I think I would prefer that: let the caller parse and move in. Just as efficient, and keeps are more informative method type.

Reply:
Why would we want that? To the caller the lockfile is just an opaque string.

Reply:
Well, the caller does want to know that `toJSON()` and this are related by a round trip law, right?

I don't mind also exposing the current signature as a simple wrapper if it is easier for call sites.